app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsCalendarAdapter.java

```diff
@@ -0,0 +1,75 @@
+package com.sdpteam.connectout.event.nearbyEvents.list;
+
+import android.content.Context;
+import android.text.format.DateFormat;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.event.Event;
+import com.sdpteam.connectout.event.viewer.EventActivity;
+import com.sdpteam.connectout.profile.ProfileActivity;
+
+import java.util.List;
+
+public class EventsCalendarAdapter extends ArrayAdapter<Event> {
+
+    private final int eventListItemResource;
+
+    /**
+     * @param context               The context in which the adapter is being used.
+     * @param eventListItemResource The resource ID for the layout file of each list item.
+     * @param events                The list of events to display in the ListView.
+     */
+    public EventsCalendarAdapter(@NonNull Context context, int eventListItemResource, @NonNull List<Event> events) {
+        super(context, eventListItemResource, events);
+        this.eventListItemResource = eventListItemResource;
+    }
+
+    public EventsCalendarAdapter(@NonNull Context context, int eventListItemResource) {
+        super(context, eventListItemResource);
+        this.eventListItemResource = eventListItemResource;
+    }
+
+    /**
+     * Creating and returning the view for each list item.
+     */
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        View view = convertView;
+        if (view == null) {
+            LayoutInflater inflater = LayoutInflater.from(getContext());
+            view = inflater.inflate(eventListItemResource, parent, false);
+        }
+
+
+        final Event event = getItem(position);
+        final String orgProfileId = event.getOrganizer();
+
+        TextView eventTitle = view.findViewById(R.id.registered_event_title);
+        eventTitle.setText(event.getTitle());
+
+        TextView registeredEventTime = view.findViewById(R.id.registered_event_time);
+        // Format the date before showing it
+        registeredEventTime.setText(DateFormat.format("dd-MM-yyyy (HH:mm:ss)",
+                event.getDate()));
+
+
+        registeredEventTime.setGravity(Gravity.CENTER);
+        eventTitle.setGravity(Gravity.CENTER);
+
+        view.setOnClickListener(v -> EventActivity.openEvent(getContext(), event.getId()));
+        return view;
```

Comment:
Nice adapter, like the solution to use a custom adapter for this calendar view.
This is the way to go, especially if in the future we want to make other types of lists that look differently

Reply:
Thanks boyiiiii!!!!!!